lib/block_view_lockups.go

```diff
@@ -0,0 +1,1784 @@
+}
+
+func (bav *UtxoView) _deleteLockupYieldCurvePoint(point *LockupYieldCurvePoint) {
+	// Create a tombstone entry.
+	tombstoneLockupYieldCurvePoint := *point
+	tombstoneLockupYieldCurvePoint.isDeleted = true
+
+	// Set the LockupYieldCurvePoint as deleted in the view.
+	bav._setLockupYieldCurvePoint(&tombstoneLockupYieldCurvePoint)
+}
+
+// Get Helper Functions for LockupYieldCurvePoint
+
+func (bav *UtxoView) GetYieldCurvePointByProfilePKIDAndDurationNanoSecs(profilePKID *PKID,
+	lockupDurationNanoSecs int64) (_lockupYieldCurvePoint *LockupYieldCurvePoint) {
+	var lockupYieldCurvePoint *LockupYieldCurvePoint
+
+	// Check the view for a yield curve point.
+	if _, pointsInView := bav.PKIDToLockupYieldCurvePointKeyToLockupYieldCurvePoints[*profilePKID]; pointsInView {
+		lockupYieldCurvePointKey := (&LockupYieldCurvePoint{
+			ProfilePKID:            profilePKID,
+			LockupDurationNanoSecs: lockupDurationNanoSecs,
+		}).ToMapKey()
+		if inMemoryYieldCurvePoint, pointExists :=
+			bav.PKIDToLockupYieldCurvePointKeyToLockupYieldCurvePoints[*profilePKID][lockupYieldCurvePointKey]; pointExists {
+			return inMemoryYieldCurvePoint
+		}
+	}
+
+	// No mapping exists in the view, check for an entry in the DB.
+	lockupYieldCurvePoint = DBGetYieldCurvePointsByProfilePKIDAndDurationNanoSecs(bav.GetDbAdapter().badgerDb,
+		bav.Snapshot, profilePKID, lockupDurationNanoSecs)
+
+	// Cache the DB entry in the in-memory map.
+	if lockupYieldCurvePoint != nil {
+		bav._setLockupYieldCurvePoint(lockupYieldCurvePoint)
+	}
+
+	return lockupYieldCurvePoint
+}
+
+func (bav *UtxoView) GetLocalYieldCurvePoints(profilePKID *PKID, lockupDuration int64) (
+	_leftLockupPoint *LockupYieldCurvePoint, _rightLockupPoint *LockupYieldCurvePoint) {
+	var leftLockupPoint *LockupYieldCurvePoint
+	var rightLockupPoint *LockupYieldCurvePoint
+
+	// Check the view for yield curve points.
+	if _, pointsInView := bav.PKIDToLockupYieldCurvePointKeyToLockupYieldCurvePoints[*profilePKID]; pointsInView {
+		for _, lockupYieldCurvePoint := range bav.PKIDToLockupYieldCurvePointKeyToLockupYieldCurvePoints[*profilePKID] {
+			// Check for nil pointer cases.
+			if lockupYieldCurvePoint.LockupDurationNanoSecs < lockupDuration && leftLockupPoint == nil {
+				leftLockupPoint = lockupYieldCurvePoint
+			}
+			if lockupYieldCurvePoint.LockupDurationNanoSecs >= lockupDuration && rightLockupPoint == nil {
+				rightLockupPoint = lockupYieldCurvePoint
+			}
+
+			// Check if the point is "more left" than the current left point.
+			if lockupYieldCurvePoint.LockupDurationNanoSecs < lockupDuration &&
+				lockupYieldCurvePoint.LockupDurationNanoSecs > leftLockupPoint.LockupDurationNanoSecs {
+				leftLockupPoint = lockupYieldCurvePoint.Copy()
+			}
+
+			// Check if the point is "more right" than the current right point.
+			if lockupYieldCurvePoint.LockupDurationNanoSecs >= lockupDuration &&
+				lockupYieldCurvePoint.LockupDurationNanoSecs < leftLockupPoint.LockupDurationNanoSecs {
```

Comment:
I think this is a bug. I think you meant `< rightLockupPoint.LockupDurationNanoSecs`

Reply:
Ah yes subtle error.